units/src/amount.rs

```diff
@@ -107,15 +107,25 @@ impl fmt::Display for Denomination {
 impl FromStr for Denomination {
     type Err = ParseDenominationError;
 
-    /// Convert from a str to Denomination.
+    /// Converts from a str to a Denomination.
     ///
     /// # Accepted Denominations
     ///
     /// All upper or lower case, excluding SI prefix (c, m, u) which must be lower case.
     /// - Singular: BTC, cBTC, mBTC, uBTC
     /// - Plural or singular: sat, satoshi, bit
     ///
-    /// Due to ambiguity between mega and milli we prohibit usage of leading capital 'M'.
+    /// Due to ambiguity between mega and milli we prohibit usage of leading capital 'M'.  It is
+    /// more important to protect users from incorrectly using a capital M to mean milli than to
+    /// allow Megabitcoin which is not a realistic denomination, and Megasatoshi which is
+    /// equivalent to cBTC which is allowed.
+    ///
+    /// # Errors
+    ///
+    /// - If the denomination begins with a capital `M` a `PossiblyConfusingDenominationError` is
```

Comment:
Is it possible to doclink?

```suggestion
    /// - If the denomination begins with a capital `M` a [`PossiblyConfusingDenominationError`] is
```

Reply:
Done